application/comit_node/src/swap_protocols/rfc003/alice/spawner.rs

```diff
@@ -75,17 +74,27 @@ impl<T: MetadataStore<SwapId>, S: StateStore<SwapId>, C: comit_client::Client> A
             secret: self.seed.new_secret(id),
         };
 
-        let state_machine_future = Alice::<AL, BL, AA, BA>::new_state_machine(
-            initiation,
-            self.ledger_events.create_ledger_events(),
-            self.ledger_events.create_ledger_events(),
-            Arc::clone(&self.comit_client_factory),
-            swap_request.bob_socket_address,
-            save_state,
-        );
+        let swap_execution = {
+            let ledger_events = self.ledger_events.clone();
+            self.comit_client_factory
+                .client_for(swap_request.bob_socket_address)
+                .then(move |client| {
```

Comment:
:question: Why is this not just an `and_then`? Then you don't have to match on the result.